packages/pages-shared/src/metadata-generator/parseRedirects.ts

```diff
@@ -59,12 +67,29 @@ export function parseRedirects(input: string): ParsedRedirects {
 			});
 			continue;
 		}
-		const from = fromResult[0];
+		let from = fromResult[0];
+		const queryParams = tokens.slice(1, index);
+		const hasInvalidQueryParam = queryParams.some((token) =>
+			token.includes("&")
+		);
+		if (hasInvalidQueryParam) {
+			invalid.push({
+				line,
+				lineNumber: i + 1,
+				message: "Query parameters cannot contain '&'.",
+			});
+			continue;
+		}
+
+		if (queryParams.length) {
+			from = `${from}?${queryParams.sort().join("&")}`;
```

Comment:
This is going to have slightly unexpected behavior if the rules has an `=` in it.

`/redirect query= /me`

Will match `/redirect?query=`, but not `/redirect?query`. I think we should normalize them here in metadata-generator to be either `x=y` or just `z` but either error-out with `z=` or silently drop the trailing `=`.

Reply:
Adding a change to drop the trailing `=` for now, happy to error if preferred